src/Button/types.ts

```diff
@@ -45,7 +45,7 @@ export type IconButtonProps = {
    * This is to be used if it is an icon-only button. Will make text visually hidden
    */
   icon: React.FunctionComponent<IconProps>
-  iconLabel: string
+  'aria-label': string
```

Comment:
Should be this required? Also, what would the user do if they wanted to apply `aria-labelledby` instead, should we have some logic to only apply the `aria-label` if `aria-labelledby` hasn't been passed in so that takes precedence?

Reply:
Related: What's the right way to add a tooltip to an IconButton, like the one on a file?

<img width="359" alt="image" src="https://user-images.githubusercontent.com/1863771/157646275-aab736a4-489f-46e4-8d7d-147084cd4c75.png">

When composable Tooltip + IconButton, does the aria-label go on both of them or just one?

<img width="495" alt="image" src="https://user-images.githubusercontent.com/1863771/157646216-f6cc6e4b-284f-421c-bdea-b65b176e27d4.png">

cc @alliethu

Reply:
@siddharthkp - I actually think we'd want the `aria-label` on the button but not the tooltip.

Here's what I think would give a nice experience for screenreaders

### When the tooltip is acting like a label
```jsx
<button aria-label="Edit this file"><svg>{/* paths and stuff */}</svg></button>
<div>Edit this file</div> // <--- this is a tooltip
```
OR
```jsx
<button aria-labelledby="editTooltip"><svg>{/* paths and stuff */}</svg></button>
<div id="editTooltip">Edit this file</div> // <--- this is a tooltip
```

### When the tooltip is acting like helper/caption text
```jsx
<button aria-label="Edit this file" aria-describedby="editTooltip"><svg>{/* paths and stuff */}</svg></button>
<div id="editTooltip">Only people in this organization can edit this file</div> // <--- this is a tooltip
```

@alliethu - please keep me correct

Reply:
Also, I just realized our tooltips aren't displayed on focus, only hover. :(

We probably need an a11y audit on these

Reply:
@mperrotti Yes, the `aria-label` should be on the `button`. Also, heads-up that @khiga8 just did a recent release to fix tooltips on our comment box toolbars. Hopefully, you should be able to reference her implementation for the fixes on the PRC end. 

https://github.com/github/accessibility/issues/594#issuecomment-1034219464

Reply:
@mperrotti even more current, @hectahertz is working on Upstreaming tooltip to PVC. The two of you should coordinate! https://github.com/github/accessibility/issues/802

Reply:
Here is the [upstreamed tooltip component](https://primer.style/view-components/components/alpha/tooltip).

Reply:
For this PR, I'll try and support both `aria-label` and `aria-labelled-by` in the types. Hope that works 🤞